xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/Window.h

```diff
@@ -166,6 +166,232 @@ namespace gui
+    ///
+    /// @return                         Focused control id
+    ///
+    ///
+    int GetFocusId()
+    {
+      return m_interface->kodi_gui->window->get_focus_id(m_interface->kodiBase, m_controlHandle);
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief To set the used label on given control id
+    ///
+    /// @param[in] controlId            Control id where label need to set
+    /// @param[in] label                Label to use
+    ///
+    ///
+    void SetControlLabel(int controlId, const std::string& label)
+    {
+      m_interface->kodi_gui->window->set_control_label(m_interface->kodiBase, m_controlHandle, controlId, label.c_str());
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief Sets a window property, similar to an infolabel.
+    ///
+    /// @param[in] key                  string - property name.
+    /// @param[in] value                string or unicode - value of property.
+    ///
+    /// @note  Key is NOT case sensitive. Setting value to an empty string is
+    ///        equivalent to clearProperty(key).\n
+    ///        You can use the above as keywords for arguments and skip certain
+    ///        optional arguments.\n
+    ///        Once you use a keyword, all following arguments require the keyword.
+    ///
+    void SetProperty(const std::string& key, const std::string& value)
+    {
+      m_interface->kodi_gui->window->set_property(m_interface->kodiBase, m_controlHandle, key.c_str(), value.c_str());
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_CWindow
+    /// @brief Returns a window property as a string, similar to an infolabel.
+    ///
+    /// @param[in] key                  string - property name.
+    /// @return                         The property as strin (if present)
+    ///
+    /// @note  Key is NOT case sensitive. Setting value to an empty string is
+    ///        equivalent to clearProperty(key).\n
+    ///        You can use the above as keywords for arguments and skip certain
+    ///        optional arguments.\n
+    ///        Once you use a keyword, all following arguments require the keyword.
+    ///
+    ///
+    std::string GetProperty(const std::string& key) const
+    {
+      std::string label;
+      char* ret = m_interface->kodi_gui->window->get_property(m_interface->kodiBase, m_controlHandle, key.c_str());
+      if (ret != nullptr)
+      {
+        if (std::strlen(ret))
+          label = ret;
```

Comment:
Is a bit to prevent the C++ class construction and copy from the not present string, not complete sure what becomes done in compiled code, but thought why let C++ work if C nothing have.